core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -239,10 +240,10 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
           }
           totalCoreCount.addAndGet(-executorInfo.totalCores)
           totalRegisteredExecutors.addAndGet(-1)
-          scheduler.executorLost(executorId, SlaveLost(reason))
+          scheduler.executorLost(executorId, reason)
           listenerBus.post(
-            SparkListenerExecutorRemoved(System.currentTimeMillis(), executorId, reason))
-        case None => logInfo(s"Asked to remove non-existent executor $executorId")
+            SparkListenerExecutorRemoved(System.currentTimeMillis(), executorId, reason.toString))
+        case None => logError(s"Asked to remove non-existent executor $executorId")
```

Comment:
Oh I think this is a bad merge conflict resolution. I'll change back to logInfo.